### PR TITLE
sFlow Truncation support

### PR DESCRIPTION
**Pull request by rck-innovium**
Support truncation for SAI\_SAMPLEPACKET\_TYPE\_SLOW\_PATH.

### CODE REVIEW

File: inc/saisamplepacket.h [hunk 1]
```diff
@@ -114,6 +114,26 @@ typedef enum _sai_samplepacket_attr_t
...
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_SAMPLEPACKET_ATTR_TRUNCATE_ENABLE,
+
+    /**
+     * @brief Truncate size. Truncate sampled packets to this size to reduce traffic bandwidth
+     *
+     * This value takes effect when SAI_SAMPLEPACKET_ATTR_TRUNCATE_ENABLE is True.
```
Comment (marian-pritsak) [inline inc/saisamplepacket.h:129]: Use @condition annotation for this

**Review by marian-pritsak — CHANGES_REQUESTED** [review]

File: doc/Samplepacket/SAI-Proposal-Samplepacket_truncation.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
++    /**
++     * @brief Truncate size. Truncate sampled packets to this size to reduce traffic bandwidth
++     *
++     * This value takes effect when SAI_SAMPLEPACKET_ATTR_TRUNCATE_ENABLE is True.
++     *
++     * @type sai_uint32_t
++     * @flags CREATE_AND_SET
++     * @default internal
++     */
++    SAI_SAMPLEPACKET_ATTR_TRUNCATE_SIZE,
```
Comment (rck-innovium) [inline doc/Samplepacket/SAI-Proposal-Samplepacket_truncation.md:123]: Qn from @JaiOCP during review on Aug-15-2024:  In SAI\_SAMPLEPACKET\_TYPE\_MIRROR\_SESSION, define the behavior when both SAI\_SAMPLEPACKET\_ATTR\_TRUNCATE\_SIZE and SAI\_MIRROR\_SESSION\_ATTR\_TRUNCATE\_SIZE are given. 

The explanation was to follow #1082: Use SAI\_SAMPLEPACKET\_ATTR\_TRUNCATE\_SIZE if we want to have the truncation specific to a port, and use SAI\_MIRROR\_SESSION\_ATTR\_TRUNCATE\_SIZE if we want the same truncation for all the ports using the same mirror session object.

Jai wanted to do more investigations and get back if the above behavior is agreeable.

File: inc/saisamplepacket.h (hunk 1 above)
Comment (rck-innovium) [reply inc/saisamplepacket.h:129]: @marian-pritsak  Done

**Review by kcudnik — APPROVED** [review]

**Review by marian-pritsak — APPROVED** [review]

**Review by JaiOCP — APPROVED** [review]
Looks Good

**Review by rlhui — APPROVED** [review]

**Pull request merged**